app/controllers/topics_controller.rb

```diff
@@ -848,7 +848,23 @@ def bulk
   end
 
   def reset_new
-    current_user.user_stat.update_column(:new_since, Time.now)
+    if params[:category_id].present?
+      category_ids = [params[:category_id]]
+
+      if params[:include_subcategories] == 'true'
```

Comment:
Why isn't `include_subcategories` a boolean?

Reply:
I think this is rails shanigans, we do this elsewhere in a few places in the same file https://github.com/discourse/discourse/blob/1c179177e7c84362572481e61b18597f7d2d5050/app/controllers/topics_controller.rb#L221-L221